### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by slava-moiseev-kramatorsk**

### CODE REVIEW

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 1]
```diff
@@ -1,5 +1,9 @@
 package mate.academy.service;
 
+import static mate.academy.service.UserService.users;
+
+import mate.academy.model.User;
+
 public class AuthenticationService {
```
Comment (Ivan95kos) [inline src/main/java/mate/academy/service/AuthenticationService.java:7]: You should use the `UserService` and method `findByEmail` here

File: src/main/java/mate/academy/service/UserService.java [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import mate.academy.model.User;
 
 public class UserService {
-    private static final User[] users = new User[] {
+    public static final User[] users = new User[] {
```
Comment (Ivan95kos) [inline src/main/java/mate/academy/service/UserService.java:6]:
```suggestion
    private static final User[] users = new User[] {
```

**Review by Ivan95kos — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/service/AuthenticationService.java (hunk 1 above)
Comment (slava-moiseev-kramatorsk) [reply src/main/java/mate/academy/service/AuthenticationService.java:7]: "Your task is to implement a login method in AuthenticationService and findByEmail method in UserService. More details are described in comments over the methods. "  -  this condition in file "README.md".
 I thought I should implement the method body in class, were method create.

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 3]
```diff
@@ -11,6 +13,16 @@ public class AuthenticationService {
      * Return false in any other cases.
      */
     public boolean login(String email, String password) {
-        return false;
+        UserService users = new UserService();
+        boolean result = false;
+        for (User user: users.getUsers()) {
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/service/AuthenticationService.java:18]: use the UserService and method findByEmail here

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 4]
```diff
@@ -11,6 +11,14 @@ public class AuthenticationService {
      * Return false in any other cases.
      */
     public boolean login(String email, String password) {
-        return false;
+        UserService users = new UserService();
```
Comment (okuzan) [inline src/main/java/mate/academy/service/AuthenticationService.java:14]:
```suggestion
        UserService userService = new UserService();
```
you have User model and UserService service, naming shouldn't be confusing

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 5]
```diff
@@ -11,6 +11,14 @@ public class AuthenticationService {
...
+        boolean result = false;
+        if (users.findByEmail(email) == null) {
+            return false;
+        }
+        if (password.equals(users.findByEmail(email).getPassword())) {
```
Comment (okuzan) [inline src/main/java/mate/academy/service/AuthenticationService.java:19]: in this impl you search for the user twice. Instead, search once and save the result to a User user local variable

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 6]
```diff
@@ -11,6 +11,14 @@ public class AuthenticationService {
...
     public boolean login(String email, String password) {
-        return false;
+        UserService userService = new UserService();
+        boolean result = false;
+        if (userService.findByEmail(email) != null
+                && (password.equals(userService.findByEmail(email).getPassword()))) {
+            return true;
+        } else {
+            result = false;
+        }
```
Comment (okuzan) [inline src/main/java/mate/academy/service/AuthenticationService.java:21]: two points not fixed - you should call findByEmail only once, and you don't need to create a local boolean variable, you can return directly

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/service/UserService.java [hunk 7]
```diff
@@ -15,6 +15,11 @@ public class UserService {
      * Return <code>null</code> if there is no suitable user
      */
     public User findByEmail(String email) {
+        for (User user: users) {
```
Comment (okuzan) [inline src/main/java/mate/academy/service/UserService.java:18]:
```suggestion
        for (User user : users) {
```

File: src/main/java/mate/academy/service/AuthenticationService.java [hunk 8]
```diff
@@ -11,6 +13,8 @@ public class AuthenticationService {
      * Return false in any other cases.
      */
     public boolean login(String email, String password) {
-        return false;
+        UserService userService = new UserService();
+        User user = userService.findByEmail(email);
+        return user != null && (user.getPassword().equals(password));
```
Comment (okuzan) [inline src/main/java/mate/academy/service/AuthenticationService.java:18]:
```suggestion
        return user != null && user.getPassword().equals(password);
```

**Review by okuzan — APPROVED** [review]
Well done!

**Pull request closed without merging**